### PR TITLE
chore: Remove `shelljs` use from backup command

### PR DESCRIPTION
**Pull request by sharat87**
This is towards fixing `appsmithctl` into a modern component of the project. First order of business is to get rid of the `shelljs` dependency, since we don't really need it that seriously, and since it does weird stuff with `require()` that we can't use `esbuild` to build `appsmithctl`.

Further PRs will remove `shelljs` from other commands as well, and then we'll remove `shelljs` from our dependencies.

## Automation

/test sanity

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11814325698>
> Commit: 98261ee8c8a52090b1e9bb27a27be4be0874b83f
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11814325698&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Wed, 13 Nov 2024 09:58:36 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Improved backup process with enhanced error handling and asynchronous checks for available disk space.
	- New function for executing commands with output capture added to utility functions.

- **Bug Fixes**
	- Refined error messages for insufficient disk space and password mismatches during backup.

- **Tests**
	- Expanded test suite for backup utilities and command execution, ensuring robust coverage of various scenarios.

- **Documentation**
	- Updated formatting of numeric constants for better readability.

### CODE REVIEW

File: deploy/docker/fs/opt/appsmith/utils/bin/utils.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+});
+
+describe("execCommandSilent", () => {
+
+  test("Runs a command", async () => {
+    await utils.execCommandSilent(["echo"]);
+  });
+
+});
```
[the remark this answers is not in the document]
Comment (sharat87) [reply deploy/docker/fs/opt/appsmith/utils/bin/utils.test.js:39]: That comes later.

File: deploy/docker/fs/opt/appsmith/utils/bin/backup.js [hunk 2]
```diff
@@ -232,13 +232,14 @@ function getTimeStampInISO() {
...
-  return parseInt(shell.exec('df --output=avail -B 1 /appsmith-stacks | tail -n 1'), 10)
+async function getAvailableBackupSpaceInBytes(path) {
+  const stat = await fsPromises.statfs(path);
+  return stat.bsize * stat.bfree;
 }
 
 function checkAvailableBackupSpace(availSpaceInBytes) {
   if (availSpaceInBytes < Constants.MIN_REQUIRED_DISK_SPACE_IN_BYTES) {
-    throw new Error('Not enough space avaliable at /appsmith-stacks. Please ensure availability of atleast 2GB to backup successfully.');
+    throw new Error("Not enough space available at /appsmith-stacks. Please ensure availability of at least 2GB to backup successfully.");
```
[the remark this answers is not in the document]
Comment (sharat87) [reply deploy/docker/fs/opt/appsmith/utils/bin/backup.js:242]: This is okay, don't have to complicate this too much.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**